File: petties-web/src/services/api/notificationService.ts
import { apiClient } from './client'

export interface ClinicNotification {
  notificationId: string
  clinicId?: string
  clinicName?: string
  type: 'APPROVED' | 'REJECTED' | 'PENDING' | 'CLINIC_PENDING_APPROVAL' | 'STAFF_SHIFT_ASSIGNED' | 'STAFF_SHIFT_UPDATED' | 'STAFF_SHIFT_DELETED'
  message: string
  reason?: string
  read: boolean
  createdAt: string
  // StaffShift-related fields
  shiftId?: string
  shiftDate?: string
  shiftStartTime?: string
  shiftEndTime?: string
}

export interface NotificationListResponse {
  content: ClinicNotification[]
  totalElements: number
  totalPages: number
  number: number
  size: number
  first: boolean
  last: boolean
}

export const notificationService = {
  /**
   * Get notifications for current user
   */
  getNotifications: async (page = 0, size = 20): Promise<NotificationListResponse> => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
    })
    const response = await apiClient.get<NotificationListResponse>(`/notifications/me?${params.toString()}`)
    return response.data
  },

  /**
   * Get unread notifications count
   */
  getUnreadCount: async (): Promise<number> => {
    const response = await apiClient.get<{ count: number }>('/notifications/me/unread-count')
    return response.data.count
  },

  /**
   * Mark notification as read
   */
  markAsRead: async (notificationId: string): Promise<void> => {
    await apiClient.put(`/notifications/${notificationId}/read`)
  },

  /**
   * Mark all notifications as read
   */
  markAllAsRead: async (): Promise<void> => {
    await apiClient.put('/notifications/me/mark-all-read')
  },
}

File: petties_mobile/lib/data/services/notification_service.dart
import '../models/notification.dart';
import 'api_client.dart';

class NotificationService {
  final ApiClient _apiClient;

  NotificationService({ApiClient? apiClient})
      : _apiClient = apiClient ?? ApiClient();

  /// Get notifications with pagination
  Future<List<NotificationModel>> getNotifications({
    int page = 0,
    int size = 20,
  }) async {
    try {
      final response = await _apiClient.get(
        '/notifications/me',
        queryParameters: {
          'page': page,
          'size': size,
        },
      );

      final List<dynamic> content = response.data['content'];
      return content.map((json) => NotificationModel.fromJson(json)).toList();
    } catch (e) {
      rethrow;
    }
  }

  /// Get unread count
  Future<int> getUnreadCount() async {
    try {
      final response = await _apiClient.get('/notifications/me/unread-count');
      return response.data['count'] as int;
    } catch (e) {
      rethrow;
    }
  }

  /// Mark notification as read
  Future<void> markAsRead(String notificationId) async {
    try {
      await _apiClient.put('/notifications/$notificationId/read');
    } catch (e) {
      rethrow;
    }
  }

  /// Mark all notifications as read
  Future<void> markAllAsRead() async {
    try {
      await _apiClient.put('/notifications/me/mark-all-read');
    } catch (e) {
      rethrow;
    }
  }
}
